### PR TITLE
Add snapcraft.yaml

### PR DESCRIPTION
**Pull request by Nazgolze**
Adds snapcraft.yaml for creating snap builds

### CODE REVIEW

File: snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    build-packages:
+      - git
+      - curl
+      - jq
+    plugin: nodejs
+    nodejs-version: 12.22.1
+    nodejs-package-manager: npm
+    source: .
+    override-build: |
+      snapcraftctl set-version "$(jq -r .version < package.json)"
```
Comment (fchavant) [inline snapcraft.yaml:22]: Those steps do not quite match those in the Platform Handbook. Where are they from?

**Review by ManuGowda — DISMISSED** [review]
LGTM, is there a way to try this with some steps?

**Comment by shuse2** [comment]
@Nazgolze  can you branch off from release/3.0.0 so that we can include this in the testnet release?

**Comment by Nazgolze** [comment]
> @Nazgolze can you branch off from release/3.0.0 so that we can include this in the testnet release?

done :+1:

**Comment by Nazgolze** [comment]
> LGTM, is there a way to try this with some steps?

@ManuGowda @shuse2 I'll either add something to readme or add a new snapcraft.md file.

File: SNAPCRAFT.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+If it's your first time running `lxd` though, you'll have to run `lxd init` to set it up (follow the prompts)
+
+If successful, you will have a newly created snap file.
+
+## Install the snap locally
+
+```
+snap install --dangerous name_of_snap.snap
+```
```
Comment (fchavant) [inline SNAPCRAFT.md:26]: There should be a note explaining why `--dangerous` is needed.

**Review by fchavant — APPROVED** [review]

**Review by ManuGowda — APPROVED** [review]

**Review by shuse2 — APPROVED** [review]

**Pull request merged**